examples/llm_stack/embedding_module.py

```diff
@@ -0,0 +1,92 @@
+from types import ModuleType
+
+import cohere
+import numpy as np
+import openai
+from sentence_transformers import SentenceTransformer
+
+from hamilton.function_modifiers import config, extract_fields
+
+
+@extract_fields(
+    dict(
+        embedding_dimension=int,
+        embedding_metric=str,
+    )
+)
+def embedding_config(embedding_service: str, model_name: str) -> dict:
+    """Return embedding dimension and distance metric based
+    embedding service and model"""
+    if embedding_service == "openai":
+        if model_name == "text-embedding-ada-002":
+            return dict(embedding_dimension=1536, embedding_metric="cosine")
+
+    elif embedding_service == "cohere":
+        if model_name == "embed-english-light-v2.0":
+            return dict(embedding_dimension=1024, embedding_metric="cosine")
+
+    elif embedding_service == "sentence_transformer":
+        if model_name == "multi-qa-MiniLM-L6-cos-v1":
+            return dict(embedding_dimension=384, embedding_metric="cosine")
+
+    else:
+        raise ValueError(
+            f"Invalid `embedding_service` and `model_name` combination.\
+             \nReceived {embedding_service} and {model_name}"
+        )
```

Comment:
any reason to not also do @config.when here? I think that would be cleaner.

Reply:
Also could drop the `extract_fields` in favor of a TypedDict as output, and then just pass that around?

Reply:
e.g. 
```python
EmbeddingConfig = TypedDict('EmbeddingConfig', {'embedding_dimension': int, 'embedding_metric': string})

@config.when(embedding_service="openai")
def embedding_config__openai(embedding_service: str, model_name: str) -> EmbeddingConfig:
    if model_name == "text-embedding-ada-002":
            return dict(embedding_dimension=1536, embedding_metric="cosine")
    # If you support more models, you would add that here
    raise ValueError(
            f"Invalid `model_name`[{model_name}] for openai was passed."
        )

# etc
```

Reply:
Never thought about using the TypedDict! That's a much cleaner interface for functions with (too) many parameters. 

The `extract_fields` seems a bit more flexibility because it makes overriding the `embedding_metric` easier for example.